EventListener/Doctrine/UploadListener.php

```diff
@@ -2,61 +2,43 @@
 
 namespace Vich\UploaderBundle\EventListener\Doctrine;
 
-use Doctrine\Common\EventArgs;
+use Doctrine\ORM\Event\PreFlushEventArgs;
+use Doctrine\ORM\Events;
 
 /**
  * UploadListener
  *
  * Handles file uploads.
  *
+ * @author Konstantin Myakshin <koc-dp@yandex.ru>
  * @author Kévin Gomez <contact@kevingomez.fr>
  */
 class UploadListener extends BaseListener
 {
     /**
-     * The events the listener is subscribed to.
-     *
-     * @return array The array of events.
+     * {@inheritdoc}
      */
     public function getSubscribedEvents()
     {
-        return array(
-            'prePersist',
-            'preUpdate',
-        );
+        return array(Events::preFlush);
     }
 
     /**
-     * @param EventArgs $event The event.
+     * @param PreFlushEventArgs $event
      */
-    public function prePersist(EventArgs $event)
+    public function preFlush(PreFlushEventArgs $event)
```

Comment:
typehint using only the classes from Doctrine Common (and add instanceof checks internally if they have not settled on a common API for the ``preflush`` event but this should also be reported to Doctrine as events have been more or less standardized in doctrine/common, but Doctrine\Common\EventArgs should extend ``\Doctrine\Common\Persistence\Event\ManagerEventArgs`` in future versions to be consistent with other events)

preflush is not properly standardized among Doctrine projects yet (check the case of the ODM fully then btw, as this event might cause you some issues)